### PR TITLE
Feature#82 create logic for resupplies

### PR DESCRIPTION
**Pull request by lucasvidela94**
# Description
What did you implemented/Why did you implemented this:
 -  Added resupplies functionality.
 -  An user can now add more stock to a product.
 - Can add resupply with product id if not given.
 - Added Pry gem to gemfile.
 - Created models for Resupply and Product resupply
 - A product can now have many resupplies.
 - Style in tailwind for resupply form.

  
  ## Screenshots 
Screenshot_20220810_005828
Screenshot_20220810_005848
image

Screenshot_20220811_132648

### CODE REVIEW

**Review by AgustinPalmaM — APPROVED** [review]
Great!

File: Gemfile [hunk 1]
```diff
@@ -52,6 +52,7 @@ gem 'dotenv-rails', groups: [:development, :test]
 gem "interactor-rails", "~> 2.0"
 
 group :development, :test do
+  gem 'pry', '~> 0.13.1'
   gem "debug", platforms: %i[ mri mingw x64_mingw ]
   gem 'rspec-rails', '~> 5.1', '>= 5.1.2'
```
Comment (JuanequeX) [inline Gemfile:57]: Please sort this gems alphabetically

File: app/controllers/resupplies_controller.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    @resupplies = Resupply.all
+    @Product = Product.all
+  end
+
+  def new 
+    @resupply = Resupply.new
+    $product_url = params[:product_id]
+  end
+
+
```
Comment (JuanequeX) [inline app/controllers/resupplies_controller.rb:15]: Delete this extra space

File: app/helpers/resupplies_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module ResuppliesHelper
+end
```
Comment (JuanequeX) [inline app/helpers/resupplies_helper.rb:2]: If, this helper is not necesary, please delete it.

File: app/models/ability.rb [hunk 4]
```diff
@@ -26,8 +26,10 @@ def initialize(user, current_business = nil)
     can :create_and_enroll, User
     can :enroll_existing_user, User
     can :remove_employee, User
+
```
Comment (JuanequeX) [inline app/models/ability.rb:29]: Delete this extra space

File: app/views/products/_form.html.erb [hunk 5]
```diff
@@ -3,7 +3,7 @@
   <div class="mb-6">
     <%= f.label 'Code *',
       for: 'product_code',
-      class: "block mb-2 text-sm font-medium text-gray-900" %>
+      class: "block mb-2 text-sm font-medi|m text-gray-900" %>
```
Comment (JuanequeX) [inline app/views/products/_form.html.erb:6]: fix this style!

File: app/views/resupplies/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+        <% if notice.present? %>
+          <p class="py-2 px-3  bg-green-50 mb-5 text-green-500 font-medium rounded-lg inline-block" id="notice">
+            <%= notice %>
+          </p>
+        <% end %>
+         <%= render 'resupply' %>
+      </div>
+    </div>
+  </div>
+</div>
```
Comment (JuanequeX) [inline app/views/resupplies/show.html.erb:18]: plase add a blank line here!

**Review by JuanequeX — CHANGES_REQUESTED** [review]

File: Gemfile (hunk 1 above)
Comment (lucasvidela94) [reply Gemfile]: done :+1:

File: app/helpers/resupplies_helper.rb (hunk 3 above)
Comment (lucasvidela94) [reply app/helpers/resupplies_helper.rb:2]: Done

File: app/views/products/_form.html.erb (hunk 5 above)
Comment (lucasvidela94) [reply app/views/products/_form.html.erb]: Already done!

File: app/models/ability.rb (hunk 4 above)
Comment (lucasvidela94) [reply app/models/ability.rb:29]: Already done!

File: app/controllers/resupplies_controller.rb (hunk 2 above)
Comment (lucasvidela94) [reply app/controllers/resupplies_controller.rb]: Done!

**Review by JuanequeX — APPROVED** [review]
LGTM

**Pull request closed without merging**